Fixes issue #153

Laike-Endaril opened the pull request:
This fixes bound scrolls not allowing players to teleport to waystones they have not activated.  The method used is to make the bound scroll activate the waystone immediately before teleporting the player, but AFTER checking that the waystone exists and shrinking the ItemStack.  See source for details.

Laike-Endaril committed:
Fixes issue #153

BlayTheNinth commented on src/main/java/net/blay09/mods/waystones/item/ItemBoundScroll.java:
```diff
@@ -105,18 +105,32 @@ public EnumActionResult onItemUseFirst(EntityPlayer player, World world, BlockPo
+                TileWaystone tileWaystone = WaystoneManager.getWaystoneInWorld(boundTo);
+                if (tileWaystone != null) //If it's going to fail, it should fail here.  If it fails inside this block, I (Laike Endaril) messed up Blay's mod :P
+                {
+                    if (!player.capabilities.isCreativeMode) itemStack.shrink(1);
```
This should stay inside the `teleportToWaystone`-if-block so that just in case the teleport does unexpectedly fail, the item won't be used up.

Laike-Endaril replied:
Fair enough. I've also changed the code style back to what should match the rest of your project (I must've forgotten to do so after editing the first time).

blay09 commented:
Please don't change the formatting of the code, it makes it hard to see the relevant changes in the pull request and results in inconsistent formatting with the rest of the code base.

Laike-Endaril committed:
Moved stack shrink into final teleport check block
Removed comment and error line because they're no longer relevant
Changed code style to what I think should match the rest of the project

Laike-Endaril commented:
Fair enough.  I also changed the code style back to what should match the rest of your project (I must've forgotten to do so after editing the first time).

BlayTheNinth committed:
Fix formatting

Laike-Endaril commented:
Woops, missed the first parenthesis of the method...maybe I should make a 2nd auto-formatting template for doing things like this.
